packages/json/src/browser/json-client-contribution.ts

```diff
@@ -42,18 +43,22 @@ export class JsonClientContribution extends BaseLanguageClientContribution {
         @inject(Languages) protected readonly languages: Languages,
         @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory,
         @inject(JsonPreferences) protected readonly preferences: JsonPreferences,
-        @inject(JsonSchemaStore) protected readonly jsonSchemaStore: JsonSchemaStore
-    ) {
+        @inject(JsonSchemaStore) protected readonly jsonSchemaStore: JsonSchemaStore,
+        @inject(JsonValidationContributionsRegistry) protected readonly jsonValidationContributionsRegistry: JsonValidationContributionsRegistry
+) {
         super(workspace, languages, languageClientFactory);
         this.initializeJsonSchemaAssociations();
     }
 
     protected updateSchemas(client: ILanguageClient): void {
-        const allConfigs = [...this.jsonSchemaStore.getJsonSchemaConfigurations()];
-        const config = this.preferences['json.schemas'];
-        if (config instanceof Array) {
-            allConfigs.push(...config);
-        }
+        const schemaStoreConfigs = this.jsonSchemaStore.getJsonSchemaConfigurations();
+        const pluginContributionConfigs = this.jsonValidationContributionsRegistry.getJsonValidations();
```

Comment:
such configs can be registered dynamically, i.e. a user install or uninstall new extension. registry should fire an event and contribution should trigger `updateSchemas` then